pandas/src/algos_groupby_helper.pxi.in

```diff
@@ -669,6 +669,72 @@ def group_cumsum(numeric[:, :] out,
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
+def group_cummin(numeric[:, :] out,
+                 numeric[:, :] values,
+                 int64_t[:] labels,
+                 numeric[:, :] accum):
+    """
+    Only transforms on axis=0
+    """
+    cdef:
+        Py_ssize_t i, j, N, K, size
+        numeric val, min_val
+        int64_t lab
+
+    N, K = (<object> values).shape
+    accum = np.empty_like(accum)
+    accum[:] = _int64_max
```

Comment:
This value (`_int64_max`) needs to be type specific - e.g., here's a failing test case

```
In [7]: df = pd.DataFrame({'key':[1], 'value': 1e99})

In [8]: df.groupby('key').cummin()
Out[8]: 
           value
0  1.104865e-311
```